llama_index/vector_stores/pinecone.py

```diff
@@ -10,6 +10,9 @@
 from functools import partial
 from typing import Any, Callable, Dict, List, Optional, cast
 
+from packaging import version
+from pkg_resources import get_distribution
```

Comment:
isn't `pkg_resources` deprecated? I've been meaning to remove an use of it from our codebase

Reply:
(Of note, though, is that LlamaIndex is still on `3.8`, so I don't know how much this rly matters.)

Reply:
@DosticJelena actually, we support 3.8 to 3.12 technically

Reply:
Actual python version is [here](https://github.com/run-llama/llama_index/blob/17e280b2bade6b5d3e3b5af6c9502f572066c201/pyproject.toml#L60)